Add more fields to `LocalOutput`

reez opened the pull request:
<!-- Erase any parts of this template not applicable to your Pull Request. -->

### Description

Add more fields to [LocalOutput](https://docs.rs/bdk_wallet/latest/bdk_wallet/struct.LocalOutput.html)

### Notes to the reviewers
Tested via building swift bindings and using that local `bdk-swift` package in `BDKSwiftExampleWallet` and using new `LocalOutput` fields in a view.

Before:
![Simulator Screenshot - iPhone 16 Pro - 2025-02-05 at 20 15 31](https://github.com/user-attachments/assets/d1efea42-90fc-4074-a42e-c98975a188f0)

After:
![Simulator Screenshot - iPhone 16 Pro - 2025-02-05 at 20 27 44](https://github.com/user-attachments/assets/9c49a0f4-304d-45d8-abb5-75ee5c8f805b)

### Changelog notice

<!-- Notice the release manager should include in the release tag message changelog -->
<!-- See https://keepachangelog.com/en/1.0.0/ for examples -->

### Checklists

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

#### New Features:

* [ ] I've added tests for the new feature
* [ ] I've added docs for the new feature

#### Bugfixes:

* [ ] This pull request breaks the existing API
* [ ] I've added tests to reproduce the issue which are now passing
* [ ] I'm linking the issue being fixed by this PR

reez commented:
Rebased so its ready when ACK'd

thunderbiscuit reviewed (changes requested):
I can't comment directly on the line (91) because those lines have not been changed as part of this PR, but from my local testing you can now simplify the `impl From<BdkCanonicalTx<'_, Arc<BdkTransaction>, BdkConfirmationBlockTime>> for CanonicalTx` by reusing the logic you just added to transform the `BdkChainPosition` into `ChainPosition`:

(keeping the old code commented so it's easy to see/find it)

```rust
impl From<BdkCanonicalTx<'_, Arc<BdkTransaction>, BdkConfirmationBlockTime>> for CanonicalTx {
    fn from(tx: BdkCanonicalTx<'_, Arc<BdkTransaction>, BdkConfirmationBlockTime>) -> Self {
        // let chain_position = match tx.chain_position {
        //     BdkChainPosition::Confirmed {
        //         anchor,
        //         transitively,
        //     } => {
        //         let block_id = BlockId {
        //             height: anchor.block_id.height,
        //             hash: anchor.block_id.hash.to_string(),
        //         };
        //         ChainPosition::Confirmed {
        //             confirmation_block_time: ConfirmationBlockTime {
        //                 block_id,
        //                 confirmation_time: anchor.confirmation_time,
        //             },
        //             transitively: transitively.map(|t| t.to_string()),
        //         }
        //     }
        //     BdkChainPosition::Unconfirmed { last_seen } => ChainPosition::Unconfirmed {
        //         timestamp: last_seen,
        //     },
        // };

        CanonicalTx {
            transaction: Arc::new(Transaction::from(tx.tx_node.tx.as_ref().clone())),
            chain_position: tx.chain_position.into(),
        }
    }
}
```

reez commented:
> I can't comment directly on the line (91) because those lines have not been changed as part of this PR, but from my local testing you can now simplify the `impl From<BdkCanonicalTx<'_, Arc<BdkTransaction>, BdkConfirmationBlockTime>> for CanonicalTx` by reusing the logic you just added to transform the `BdkChainPosition` into `ChainPosition`:
> 
> (keeping the old code commented so it's easy to see/find it)
> 
> ```rust
> impl From<BdkCanonicalTx<'_, Arc<BdkTransaction>, BdkConfirmationBlockTime>> for CanonicalTx {
>     fn from(tx: BdkCanonicalTx<'_, Arc<BdkTransaction>, BdkConfirmationBlockTime>) -> Self {
>         // let chain_position = match tx.chain_position {
>         //     BdkChainPosition::Confirmed {
>         //         anchor,
>         //         transitively,
>         //     } => {
>         //         let block_id = BlockId {
>         //             height: anchor.block_id.height,
>         //             hash: anchor.block_id.hash.to_string(),
>         //         };
>         //         ChainPosition::Confirmed {
>         //             confirmation_block_time: ConfirmationBlockTime {
>         //                 block_id,
>         //                 confirmation_time: anchor.confirmation_time,
>         //             },
>         //             transitively: transitively.map(|t| t.to_string()),
>         //         }
>         //     }
>         //     BdkChainPosition::Unconfirmed { last_seen } => ChainPosition::Unconfirmed {
>         //         timestamp: last_seen,
>         //     },
>         // };
> 
>         CanonicalTx {
>             transaction: Arc::new(Transaction::from(tx.tx_node.tx.as_ref().clone())),
>             chain_position: tx.chain_position.into(),
>         }
>     }
> }
> ```
> 
> Just to be clear line 89 would then become:
> 
> ```rust
> impl From<BdkCanonicalTx<'_, Arc<BdkTransaction>, BdkConfirmationBlockTime>> for CanonicalTx {
>     fn from(tx: BdkCanonicalTx<'_, Arc<BdkTransaction>, BdkConfirmationBlockTime>) -> Self {
>         CanonicalTx {
>             transaction: Arc::new(Transaction::from(tx.tx_node.tx.as_ref().clone())),
>             chain_position: tx.chain_position.into(),
>         }
>     }
> }
> ```

Wow great call, thanks!

Fixed, and rebased as well.

thunderbiscuit commented:
I just merged #670, so if you rebase on that the live test won't run.

reez commented:
Ready for approval.

thunderbiscuit reviewed (approved):
ACK e3bcb4bc059956fdc2a957a99b51e9e1851cb07c.
